usersimcrs/run_simulation.py

```diff
@@ -324,8 +325,13 @@ def load_rasa_diet_classifier(
     intent_schema_file = config["intents"].get()
     intent_schema = yaml.load(open(intent_schema_file), Loader=yaml.FullLoader)
 
-    agent_intents_str = intent_schema["agent_elicit_intents"]
-    agent_intents_str.extend(intent_schema["agent_set_retrieval"])
+    agent_intents_str: Set[str] = set()
+    for v in intent_schema["user_intents"].values():
+        intents = v.get("expected_agent_intents", [])
+        if intents:
+            agent_intents_str.update(intents)
```

Comment:
alternatively (in one less line):
```
if "expected_agent_intents" in v:
    agent_intents_str.update(v["expected_agent_intents"])
```

Reply:
Good point! Changed code with suggestion.